File: SharpBot/SharpBot/Move.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SharpBot
{
    enum MoveType : int
    {
        Pass = 0,
        Attack = 1,
        Strengthen = 2,
    };

    class Move
    {
        public Move(MoveType type, BoardLocation from, BoardLocation to)
        {
            this.type = type;
            this.from = from;
            this.to = to;
        }

        private readonly MoveType type;
        public MoveType Type { get { return type; } }

        private readonly BoardLocation from;
        public BoardLocation From { get { return from; } }

        private readonly BoardLocation to;
        public BoardLocation To { get { return to; } }
    }
}

File: java-bot/src/gos/bot/protocol/Move.java
package gos.bot.protocol;

public class Move
{
    public Move(MoveType type, BoardLocation from, BoardLocation to)
    {
        Type = type;
        From = from;
        To = to;
    }
    
    public final MoveType Type;
    public final BoardLocation From;
    public final BoardLocation To;
}
